src/java/org/apache/cassandra/config/Converter.java

```diff
@@ -0,0 +1,204 @@
+ * parameter name. (CASSANDRA-15234)
+ * It is important to be noted that this converter is not intended to be used when we don't change name of a configuration
+ * parameter but we want to add unit. This would always default to the old value provided without a unit at the moment.
+ * in case this functionality is needed at some point, please, raise a ticket.
+ */
+public interface Converter<Original, Current>
+{
+    /**
+     * A method to identify what type is needed to be returned from the conversion used for a configuration parameter
+     * in {@link Replaces} annotation in {@link Config}
+     * @return class type
+     */
+    Class<Original> getInputType();
+
+    /**
+     * Apply the converter specified as part of the {@link Replaces} annotation in {@link Config}
+     * @param value we will use from cassandra.yaml to create a new {@link Config} parameter of type {@link CassandraDuration},
+     * {@link BitRate} or {@link DataStorage}
+     * @return new object of type {@link CassandraDuration}, {@link BitRate} or {@link DataStorage}
+     */
+    Current apply(Original value);
+
+    /**
+     * This converter is used when we change the name of a cassandra.yaml configuration parameter but we want to be
+     * able to still use the old name too. No units involved.
+     */
+    public static final class IdentityConverter implements Converter<Object, Object>
+    {
+        public Class<Object> getInputType()
+        {
+            return null; // null means 'unchanged'  mostly used for renames
+        }
+
+        public Object apply(Object value)
+        {
+            return value;
+        }
+    }
+
+    public static final class MillisDurationConverter implements Converter<Long, CassandraDuration>
+    {
+        public Class<Long> getInputType()
+        {
+            return Long.class;
+        }
+
+        public CassandraDuration apply(Long value)
+        {
+            if (value == null)
+                return null;
+            return CassandraDuration.inMilliseconds(value);
+        }
+    }
+
+    public static final class MillisDurationInDoubleConverter implements Converter<Double, CassandraDuration>
+    {
+
+        public Class<Double> getInputType()
+        {
+            return Double.class;
+        }
+
+        public CassandraDuration apply(Double value)
+        {
+            if (value == null)
+                return null;
+            return CassandraDuration.inMilliseconds((long)value.doubleValue());
+        }
+    }
+
+    public static final class MillisDurationConverterCustom implements Converter<Long, CassandraDuration>
+    {
+        public Class<Long> getInputType()
+        {
+            return Long.class;
+        }
+
+        public CassandraDuration apply(Long value)
+        {
+            if (value == null)
+                return null;
+
+            if (value.equals((long) -1))
```

Comment:
Would, at minimum, comment here on the fact that -1 was a legacy default value. Trying to figure out if there's a better way to model this though. (ex. A default mapping specified in `@Replaces`)

Reply:
Oh I forgot yesterday we have this covered here when we discussed those three parameters, probably added last year. I will definitely add a comment to be clear why we have it. Actually a java doc even better for that particular converter. 
But I wouldn't add more fields because of this particular thing to `@Replaces` to overwhelm it. We have custom converter and being documented should be enough IMO.